prometheus_pgbouncer_exporter/cli.py

```diff
@@ -16,23 +16,24 @@ def read_config_file(config_file):
     config = Config()
     try:
         config.read(config_file)
-        logging.getLogger().info("Config file successfully read from {file}".format(file=args.config))
+        logging.getLogger().info("Config file successfully read from {file}".format(file=config_file))
     except Exception as error:
-        logging.getLogger().fatal("Unable to read config file from {file}".format(file=args.config), extra={"exception": str(error)})
+        logging.getLogger().fatal("Unable to read config file from {file}".format(file=config_file), extra={"exception": str(error)})
         raise Exception('Cannot read config file')
 
     # Validate config
     try:
         config.validate()
     except Exception as error:
-        logging.getLogger().fatal("The config file {file} is invalid: {error}".format(file=args.config, error=str(error)))
+        logging.getLogger().fatal("The config file {file} is invalid: {error}".format(file=config_file, error=str(error)))
         raise Exception('Invalid config file')
 
     return config
 
 
 def main():
     shutdown = False
+    sighup_received_count = {'val': 0}
```

Comment:
why is this a dict?

Reply:
https://stackoverflow.com/questions/3190706/nonlocal-keyword-in-python-2-x

This has some alternatives although this hack is the top answer fwiw. Personally I don't like the function attr solution as it's even less obvious. The other suggestion is to make a class or object to hold the nonlocal.

Reply:
I didn't finish my thought :) I'd just go with the dict and add a comment about why you're doing it that way.

Reply:
Just found out this project only supports python>3, so we can use `nonlocal` in the inner function here.